lnrpc/routerrpc/router_backend.go

```diff
@@ -1191,6 +1195,46 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 		payIntent.DestFeatures = features
 	}
 
+	// Validate that the MPP parameters are compatible with the
+	// payment amount. In other words, the parameters are invalid if
+	// they do not permit sending the full payment amount.
+	if rpcPayReq.MaxShardSizeMsat > 0 {
+		if payIntent.MaxParts > MaxPartsUpperLimit {
```

Comment:
this check should be moved to where we assign `payIntent.MaxParts` at L863 so we can exit early.